### DISCUSSION TITLE
Programmatically unmount a component, to fire onCleanup lifecycle method

### DISCUSSION THREAD

**Discussion by doldsimo** [solidjs/solid#1679] [Q&A] [answered]
Hello,

I have a question, whether you can programmatically unmount a component?
By that I mean that the component then fires the "onCleanup" lifecycle method.

So, for example, can I use a method that I fire with a button click to unmount another certain component?

**Comment by snnsnn** [comment, marked as answer]
Use Show, it unmounts its children when its condition toggles off.
